jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/WebAppProvider.java

```diff
@@ -492,6 +477,16 @@ protected void fileRemoved(String filename) throws Exception
         //is the file that was removed an .xml file?
         if (lowname.endsWith(".xml"))
             super.fileRemoved(filename);
+
+        //is there a .xml file of the same name?
+        if (exists(file.getName()+".xml")||exists(file.getName()+".XML"))
+            return; //assume we will get removed events for the xml file
+
+        //is there .war file of the same name?
+        if (exists(file.getName()+".war")||exists(file.getName()+".WAR"))
```

Comment:
Same.  What if it's `.War` ?

Reply:
You could use the test from line 465 here, right?

Reply:
I don't really think we need to worry about War and wAr and waR etc.     But then I guess it would do no harm to write a utility to check the 8 permutations for existence.... but what do we then do if multiple versions of them exist?   Perhaps it is best to let sleeping dogs lie for now.